draft-ietf-rats-corim.md

```diff
@@ -2371,14 +2392,42 @@ Endorsements are added to the ACS if the Endorsement condition is satisifed by t
+Each `ev` entry is processed independently of other `ev`s.
+
+The Verifier SHALL find all sets of ACS-ECTs which match all the `ev`.`condition`s, where the ACS-ECT `cmtype` contains either `evidence`, or `endorsements`.
+For each match ({{sec-match-condition-ect}}), all the `ev`.`addition` ECTs are added to the ACS.
 
-#### Processing Conditional Endorsements {#sec-process-cond-end}
+Note that some condition values can match against multiple ACS-ECTs, or sets of ACS-ECTs.
+If there are multiple matches then each match is processed independently from the others.
 
-Conditional Endorsement Triples are transformed into an internal representation based on `ev`.
-Conditional endorsements have the same processing steps as shown in ({{sec-process-end}}).
+##### Copying instance and/or group fields from a condition {#sec-comid-instance-group-copy}
+
+A CoRIM author may need to create a conditional endorsement which applies to all measurements which have the same `class` field within their `environment-map`, regardless of their `instance` and/or `group` fields.
+The `tagged-variable` option in these fields can be used to achieve this.
+
+In the simplest case, within a CoRIM, the `enviroment-map`.`instance` field of an `ev` condition holds a `tagged-variable` and the `enviroment-map`.`instance` field of an addition within that `ev` holds the same `tagged-variable`.
+The `tagged-variable` type contains an integer `var-number` distinguishing different sets of `instance` fields.
+
+Within each triple, each `tagged-variable` can only be bound to one value, so if there are multiple `environment-map` conditions using the same `tagged-variable` then subsequent uses of `tagged-variable` with the same `var-number` introduce a constraint that all `instance`s which match a `tagged-variable` must be binary identical.
+
+After successfully matching an `ev` condition ECT whose `instance` field is a `tagged-variable` against an ACS entry, the Verifier SHALL bind the `instance` value to the `instance` value of the ACS-ECT which matched against that condition ECT.
+If the matching ACS-ECT does not include an instance then the bound `instance` value is the empty value.
+
+An `ev` addition ECT whose `enviroment-map`.`instance` field is a `tagged-variable` references the bound value.
+When creating the ECT which will be added to the ACS, if the referenced `tagged-variable` is bound to a non-empty value, then the Verifier SHALL copy that bound value to the `enviroment-map`.`instance` field of the ACS-ECT created by the `ev` addition.
+If the referenced variable is bound to the empty value then the Verifier SHALL NOT add an `instance` field.
+If the referenced variable is not bound then the triple is invalid - the Verifier SHALL NOT add the addition ECT to the ACS.
+
+If a conditional endorsement containing `ev` conditions which use `tagged-variables` matches against multiple ECTs, then each match is processed independently of the others.
```

Comment:
Such muddling is not helping, `addition` then again followed by `condition` again